stellar-dotnet-sdk-test/RevokeSponsorshipTest.cs

```diff
@@ -0,0 +1,441 @@
+        Assert.AreEqual("0.00001", transactionResult.FeeCharged);
+        var results = (List<OperationResult>)((TransactionResultSuccess)transactionResult).Results;
+        Assert.AreEqual(1, results.Count);
+        Assert.IsInstanceOfType(results[0], typeof(RevokeSponsorshipSuccess));
+
+        await Task.Delay(1000);
+        // Remove the offer so the test won't fail next run
+        account = await _server.Accounts.Account(_sponsoredId);
+
+        var removeOfferOperation = new ManageSellOfferOperation
+                .Builder(new AssetTypeNative(), _assetA, "0", "1.5", offerId)
+            .Build();
+        
+        // Remove the trustline so the test won't fail the next run
+        var removeTrustOperation =
+            new ChangeTrustOperation.Builder(_assetA, "0").SetSourceAccount(_sponsoredAccount).Build();
+        tx = new TransactionBuilder(account)
+            .AddOperation(removeOfferOperation)
+            .AddOperation(removeTrustOperation)
+            .Build();
+
+        tx.Sign(_sponsoredAccount);
+        await Task.Delay(1000);
+        
+        await _server.SubmitTransaction(tx);
+    }
+
+    private async Task RevokeTrustlineSponsorship(Asset asset)
+    {
+        var account = await _server.Accounts.Account(_sponsoringId);
+
+        var revokeOperation = new RevokeLedgerEntrySponsorshipOperation
+                .Builder(_sponsoredId, asset)
+            .Build();
+
+        var tx = new TransactionBuilder(account)
+            .AddOperation(revokeOperation)
+            .Build();
+        tx.Sign(_sponsoringAccount);
+
+        var txResponse = await _server.SubmitTransaction(tx);
+        Assert.IsNotNull(txResponse);
+        Assert.IsTrue(txResponse.IsSuccess());
+        Assert.IsNotNull(txResponse.ResultXdr);
+        var transactionResult = TransactionResult.FromXdrBase64(txResponse.ResultXdr);
+        Assert.IsTrue(transactionResult.IsSuccess);
+        Assert.IsInstanceOfType(transactionResult, typeof(TransactionResultSuccess));
+        Assert.AreEqual("0.00001", transactionResult.FeeCharged);
+        var results = (List<OperationResult>)((TransactionResultSuccess)transactionResult).Results;
+        Assert.AreEqual(1, results.Count);
+        Assert.IsInstanceOfType(results[0], typeof(RevokeSponsorshipSuccess));
+
+        await Task.Delay(1000);
+        // Remove the trustline so the test won't fail the next run
+        account = await _server.Accounts.Account(_sponsoredId);
+        // Try removing the trust line if exists
+        var removeTrustOperation =
+            new ChangeTrustOperation.Builder(asset, "0").SetSourceAccount(_sponsoredAccount).Build();
+        tx = new TransactionBuilder(account)
+            .AddOperation(removeTrustOperation)
+            .Build();
+
+        tx.Sign(_sponsoredAccount);
+        await Task.Delay(1000);
+    }
+
+    private async Task<string> CreateSponsoredClaimableBalance()
```

Comment:
why is this here? Is the name of this file maybe incorrect? Should it be SponsorshipTests? as we sponsor and revoke in the same file.

Reply:
I forgot to submit it.

My mistake, I created this file with only revoke sponsorship actions in mind.